src/OpenTelemetry.Contrib.Extensions.AWSXRay/Instrumentation/AWSTracingPipelineHandler.cs

```diff
@@ -0,0 +1,238 @@
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+        }
+
+        public override async Task<T> InvokeAsync<T>(IExecutionContext executionContext)
+        {
+            T ret = null;
+
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                ret = await base.InvokeAsync<T>(executionContext).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+
+            return ret;
+        }
+
+        private Activity ProcessBeginRequest(IExecutionContext executionContext)
+        {
+            Activity activity = null;
+
+            var requestContext = executionContext.RequestContext;
+            var service = this.GetAWSServiceName(requestContext);
+            var operation = this.GetAWSOperationName(requestContext);
+
+            activity = AWSSDKActivitySource.StartActivity(service + "." + operation, ActivityKind.Client);
+
+            if (activity == null)
+            {
+                return null;
+            }
+
+            activity.AddTag(AWSSemanticConventions.AttributeAWSServiceName, service);
+            activity.AddTag(AWSSemanticConventions.AttributeAWSOperationName, operation);
+
+            var client = executionContext.RequestContext.ClientConfig;
+            if (client != null)
+            {
+                activity.AddTag(AWSSemanticConventions.AttributeAWSRegion, client.RegionEndpoint?.SystemName);
+            }
+
+            this.AddRequestSpecificInformation(activity, requestContext, service);
+
+            AwsPropagator.Inject(new PropagationContext(activity.Context, Baggage.Current), requestContext.Request.Headers, Setter);
+
+            return activity;
+        }
+
+        private void ProcessEndRequest(IExecutionContext executionContext, Activity activity)
+        {
+            var responseContext = executionContext.ResponseContext;
+            var requestContext = executionContext.RequestContext;
+
+            if (activity.GetTagValue(AWSSemanticConventions.AttributeAWSRequestId) == null)
+            {
+                activity.AddTag(AWSSemanticConventions.AttributeAWSRequestId, this.FetchRequestId(requestContext, responseContext));
+            }
+
+            var httpResponse = responseContext.HttpResponse;
+            if (httpResponse != null)
+            {
+                this.AddStatusCodeToActivity(activity, (int)httpResponse.StatusCode);
+                activity.SetTag(AWSSemanticConventions.AttributeHttpResponseContentLength, httpResponse.ContentLength);
```

Comment:
besides the status code, normally we try to set if the request was Ok,Unset,Error. We have a herper for that (activity.SetStatus(StatusCode.Ok/Unset/Error)

Reply:
Setting the status as Unset for cases where the status is not already set to Error by `ProcessException` method.

Reply:
I think the condition should be !status.Equals instead of status.Equals.

Reply:
Right. Fixed the condition and added a status check in the unit tests as well.